src/main/java/net/earthcomputer/clientcommands/command/VillagerCommand.java

```diff
@@ -0,0 +1,309 @@
+        if (resultPredicate.value() instanceof ItemAndEnchantmentsPredicate predicate) {
+            resultItemString = predicate.toString();
+        } else {
+            try {
+                ItemParser.ItemResult resultItemResult = new ItemParser(registries).parse(new StringReader(resultPredicate.string()));
+                resultItemString = displayText(new ItemStack(resultItemResult.item(), 1, resultItemResult.components()), true);
+            } catch (CommandSyntaxException e) {
+                resultItemString = resultPredicate.string();
+            }
+        }
+
+        goals.add(new Goal(
+            String.format("%sx %s", firstItemQuantityRange.string(), firstItemString),
+            item -> firstPredicate.value().test(item) && firstItemQuantityRange.value().matches(item.getCount()),
+
+            secondPredicate == null || secondItemQuantityRange == null ? null : String.format("%sx %s", secondItemQuantityRange.string(), secondItemString),
+            secondPredicate == null || secondItemQuantityRange == null ? null : item -> secondPredicate.value().test(item) && secondItemQuantityRange.value().matches(item.getCount()),
+
+            String.format("%sx %s", resultItemQuantityRange.string(), resultItemString),
+            item -> resultPredicate.value().test(item) && resultItemQuantityRange.value().matches(item.getCount())));
+        ClientCommandHelper.sendFeedback("commands.cvillager.goalAdded");
+
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int listGoals(FabricClientCommandSource source) {
+        if (goals.isEmpty()) {
+            source.sendFeedback(Component.translatable("commands.cvillager.listGoals.noGoals").withStyle(style -> style.withColor(ChatFormatting.RED)));
+        } else {
+            if (goals.size() == 1) {
+                source.sendFeedback(Component.translatable("commands.cvillager.listGoal.success"));
+            } else {
+                source.sendFeedback(Component.translatable("commands.cvillager.listGoals.success", FishingCracker.goals.size() + 1));
+            }
+            for (int i = 0; i < goals.size(); i++) {
+                Goal goal = goals.get(i);
+                source.sendFeedback(Component.literal((i + 1) + ": " + goal.toString()));
+            }
+        }
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int removeGoal(FabricClientCommandSource source, int index) throws CommandSyntaxException {
+        index = index - 1;
+        if (index < goals.size()) {
+            Goal goal = goals.remove(index);
+            source.sendFeedback(Component.translatable("commands.cvillager.removeGoal.success", goal.toString()));
+        } else {
+            throw INVALID_GOAL_INDEX.create(index + 1, goals.size());
+        }
+        return Command.SINGLE_SUCCESS;
+    }
+
+    private static int setVillagerTarget(@Nullable Entity target) throws CommandSyntaxException {
+        if (!(target instanceof Villager) && target != null) {
```

Comment:
I think you can use pattern matching for `instanceof` here to avoid the cast to `Villager`.